mira/sources/system_dynamics/stella.py

```diff
@@ -52,37 +59,50 @@ def template_model_from_stella_model_file(fname) -> TemplateModel:
     :
         A MIRA template model
     """
+    with open(fname) as f:
+        xml_str = f.read()
 
-    pysd_model = pysd.read_xmile(fname)
-    stella_model_file = XmileFile(fname)
-    stella_model_file.parse()
-
-    expression_map = extract_stella_variable_expressions(stella_model_file)
-    return template_model_from_pysd_model(pysd_model, expression_map)
+    return template_model_stella_model_string(xml_str)
 
 
 def template_model_from_stella_model_url(url) -> TemplateModel:
-    """Return a template model from a Stella model file provided by an url.
+    """Return a template model from a Stella model file provided by an url
 
     Parameters
     ----------
     url : str
-        The url to the Stella model file.
+        The url to the Stella model file
 
     Returns
     -------
     :
         A MIRA Template Model
     """
 
-    file_extension = Path(url).suffix
-    data = requests.get(url).content
+    xml_str = requests.get(url).content.decode("utf-8")
```

Comment:
Usually, one can use `text` instead of `content` to get the decoded string from the response. Is the explicit UTF-8 decoding necessary here?